powershell-modules/AWSToolsforPowerShell/Demo-AWS.Tools/build-AWSToolsLayer.ps1

```diff
@@ -1,24 +1,63 @@
-New-Item -Path $modulesPath -ItemType Directory -Force | Out-Null
-
-Write-Host "Downloading full AWSTools module to stage area:" $stagePath -foregroundcolor "green"
-Invoke-WebRequest  -Uri https://sdk-for-net.amazonwebservices.com/ps/v4/latest/AWS.Tools.zip -OutFile $stagePath\AWS.Tools.zip
-
-### Extract entire AWS.Tools modules to stage area but only move over select AWS.Tools modules (AWS.Tools.Common required)
-Write-Host "Extracting full AWSTools module to stage area:" $stagePath -foregroundcolor "green"
-Expand-Archive $stagePath\AWS.Tools.zip $stagePath -Force
-
-Write-Host "Moving selected AWSTools modules to modules directory:"$modulesPath -foregroundcolor "green"
-Move-Item "$stagePath\AWS.Tools.Common" "$modulesPath" -Force
-# Move-Item "$stagePathe\AWS.Tools.S3" "$modulesPath" -Force
-# Move-Item "$stagePath\AWS.Tools.EventBridge" "$modulesPath" -Force
-
-Write-Host "Deleting AWSTools stage area" -foregroundcolor "green"
-Remove-Item $stagePath -Recurse
\ No newline at end of file
+<#
+    .SYNOPSIS
+    A script to create a folder to be used as a Lambda layer containing the specified AWS.Tools modules.
+#>
+param (
+    # A list of AWS.Tools modules to embed in the Lambda layer.
+    [string[]]$ModuleList = @('AWS.Tools.Common'),
+
+    # The folder path where the layer content should be created.
+    [ValidateNotNullOrEmpty()]
+    [string]$LayerPath = ([System.IO.Path]::Combine($PSScriptRoot, 'layers', 'modulesLayer')),
+
+    # The URL to the AWS.Tools zip file.
+    [string]$AWSToolsSource = 'https://sdk-for-net.amazonwebservices.com/ps/v4/latest/AWS.Tools.zip'
+)
+
+function Log {
+    param (
+        [Parameter(Position=0)]
+        $Message,
+        $ForegroundColor = 'Green'
+    )
+    Write-Host "AWSToolsLayer: $Message" -ForegroundColor $ForegroundColor
+}
+
+$ProgressPreference = 'SilentlyContinue'
+Log 'Starting to build the AWS.Tools layer.' -ForegroundColor 'Yellow'
+
+if (Test-Path -Path $LayerPath -PathType Container) {
+    $null = Remove-Item -Path $LayerPath -Recurse
+}
+
+$stagePath = Join-Path -Path $LayerPath -ChildPath 'staging'
+Log "Using the staging path: '$stagePath'."
+Log "Using the layer path: '$LayerPath'."
+$LayerPath, $stagePath | ForEach-Object {
+    $null = New-Item -Path $_ -ItemType Directory -Force
+}
+
+Log "Downloading the AWS.Tools modules from '$AWSToolsSource'."
+$zipFile = Join-Path -Path $stagePath -ChildPath 'AWS.Tools.zip'
+Invoke-WebRequest -Uri $AWSToolsSource -OutFile $zipFile
+
+Log 'Expanding the AWS.Tools zip file to the staging path.'
+Expand-Archive -Path $zipFile -DestinationPath $stagePath -Force
+
+$ModuleList | ForEach-Object {
+    Log "Moving the '$_' module to the layer path."
+    $modulePath = Join-Path -Path $stagePath -ChildPath $_
+    if (-not(Test-Path -Path $modulePath -PathType Container)) {
+        throw "Cannot find the module '$_' in the staging path '$stagePath'."
+    }
+    Move-Item -Path $modulePath -Destination $LayerPath -Force
```

Comment:
The `AWS.Tools.Common` folder is ending up in the root of the `modulesLayer` folder. It needs to be in a `modules` subfolder, otherwise the runtime can't find the modules.

Reply:
Good call. Adding pester tests to provide validations to prevent these regressions.